plugins/linuxplugin/interface_config.go

```diff
@@ -215,12 +223,21 @@ func (plugin *LinuxInterfaceConfigurator) ConfigureLinuxInterface(iface *intf.Li
 		return nil
 	}
 
+	// Switch to VEth pre-configure namespace
 	nsMgmtCtx := linuxcalls.NewNamespaceMgmtCtx()
+	revert, err := linuxcalls.SwitchNamespace(nsMgmtCtx, plugin.vethCfgNamespace)
+	if err != nil {
+		return err
+	}
+
 	err = plugin.addVethInterface(nsMgmtCtx, iface, peer.config)
 	if err != nil {
 		return err
```

Comment:
Missing revert() before return. Probably safer way is to put SwitchNamespace() into addVethInterface() and use revert with defer. This way it is automatic and never missed.

Also in addVethInterface() you may remove this code block which is now obsolete:

	// VETH is first created in the default namespace so it has to be removed there as well.
	err = plugin.removeObsoleteVeth(nsMgmtCtx, iface.Name, iface.HostIfName, nil)
	if err != nil {
		return err
	}
	err = plugin.removeObsoleteVeth(nsMgmtCtx, peer.Name, peer.HostIfName, nil)
	if err != nil {
		return err
	}

Or maybe replace nil with plugin.vethCfgNamespace if there is a change that some veth may have remained there from before (e.g. if a previous addVethInterface failed in progress)